src/editor_view.cpp

```diff
@@ -496,4 +527,31 @@ void EditorView::zoomToFit(QRectF nodeBoundingRect)
     m_nodeBoundingRect = nodeBoundingRect;
 }
 
+void EditorView::drawBackground(QPainter *painter, const QRectF &rect)
+{
+    painter->save();
+
+    painter->fillRect(rect, this->backgroundBrush());
+
+    if (m_grid_visible)
+    {
+        const int gridSize = m_grid.size();
+
+        qreal left = int(rect.left()) - (int(rect.left()) % gridSize);
```

Comment:
I believe the application will crash here if `gridSize` is zero, which it initially is. That needs to be checked maybe in the main if. 

Also left and top could be consts.